native/lib.rs

```diff
@@ -0,0 +1,45 @@
+mod droptimer;
+
+use helium_crypto::{ecc_compact, ed25519, KeyType, Verify};
+use rustler::{Atom, Binary, Error as NifError, ListIterator, NifResult};
+use std::convert::TryFrom;
+
+// -spec verify(
+//     [{Bin :: binary(), [{Signature :: binary(), PubKeyBin :: libp2p_crypto:pubkey_bin()}, ...]}]
+// ) -> ok | {error, Reason :: binary()}.
+#[rustler::nif(name = "verify", schedule = "DirtyCpu")]
+fn verify_1(batch: ListIterator) -> NifResult<Atom> {
```

Comment:
All of these checks are independent, so any parallelization that could be done here could be very valuable. If any check fails kill all the other checks immediately.

Reply:
Short circuiting a parallel operation is significantly more complicated than a serial one. About how often should verification return false?